### PR TITLE
Do not match the body tag eagerly

### LINKED ISSUE #80: update from 0.9.0 to 0.10.0 breaks emojis
I updated the gems of my Jamstack website and found out that 0.10.0 is the gem that breaks eventually the emojis.

Actual result:

The website contains `:bike:`

Expected result:

The website contains :bike: (as emoji)

Have a look here: https://github.com/rriemann/blog.riemann.cc/commit/7c3efdb4b3d7837855e10a0480296d3653745bba

The next commit reverts the jemoji update to make emojis working again.

Unfortunately, I cannot provide at the moment a minimal not-working example. I use a proprietary gem in my setup.

Edit:

https://5b444c09fdd72a6adce6dbfd--blog-riemann.netlify.com/digitalisation/2018/06/06/open-letter-call-for-a-collaborative-data-protection-faq/

The table in the link above (output of the linked commit) should have plenty country flag emojis.

### PR DESCRIPTION
**Pull request by ashmaroli**
Fixes #80 (and #81)
Fixes github/pages-gem#578

/cc @rriemann @parkr

### CODE REVIEW

**Comment by pathawks** [comment]
What went wrong here?

**Comment by ashmaroli** [comment]
> What went wrong here?

Travis issue.. A restart of the errored build *may* help..

**Comment by pathawks** [comment]
> What went wrong here?

I mean, how did we let such a bug get merged and released? What does this PR fix? What was the problem? How can we prevent this from happening again?

**Comment by ashmaroli** [comment]
> What was the problem?

A regression was introduced via #74 which overlooked the fact that the regex introduced therein could match the entire markup starting from `<body` when the HTML template was minified.
Under normal circumstances, the newline `\n` immediately following a `<body>` would terminate the match at the newline char.

Since **I never anticipated** the regex to match *overzealously*, tests were not written to verify that either. 

The current proposal does all that..

**Comment by parkr** [comment]
Is there a test for a body tag without any attributes like `<body>...</body>`?

**Comment by ashmaroli** [comment]
> Is there a test for a body tag without any attributes

I'll add one explicitly..

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -13,8 +13,15 @@ def find_by_title(docs, title)
...
 
-  def index_fixture
-    File.read File.expand_path("fixtures/index.html", __dir__)
+  %w(
+    index
+    minified
+    plain_index
+    plain_minified
+  ).each do |method_name|
+    define_method "#{method_name}_fixture".to_sym do
```
Comment (parkr) [inline spec/spec_helper.rb]: In Jekyll we do `fixture(“plain_index.html”)`. I don’t love the metaprogramming here. Could be use a method argument instead?

**Comment by parkr** [comment]
@jekyllbot: merge +fix

**Comment by parkr** [comment]
See https://github.com/github/pages-gem/pull/582.

**Pull request merged**